### ISSUE TITLE
Make the redundant freeze message less dogmatic, more helpful

### ISSUE THREAD

**Issue by brandonblack** [bbatsov/rubocop#2890]
The current error message for `Style/RedundantFreeze` feels a bit passive aggressive / dogmatic and doesn't actually help developers resolve the situation effectively. 

Calling `.freeze` on strings in a file that contains the `frozen_string_literal` declaration is in fact _**NOT**_ pointless if you're trying to be both forward and backward compatible. Those of us still supporting older versions of Ruby as well as 2.3.x+ need to keep both of these in place.

The cop should be updated to:
- Conditionally enforce the check under `when_needed` and >= 2.3
- Reword the error message to guide users to set `TargetRubyVersion` accordingly.

For example:
```ruby
MSG = 'Freezing immutable objects is has no effect. If this ' \
      'usage is an attempt to be backward compatible, set ' \
      '`TargetRubyVersion` accordingly to resolve this.'.freeze
```

Relates to: #2611

**Comment by segiddins** [comment]
It does use the target ruby version to determine if the freeze is redundant
